lib/skate/settings/user.ex

```diff
@@ -8,7 +8,7 @@ defmodule Skate.Settings.User do
   @doc """
   Get a user with the matching email, if one exists
   """
-  def get_by_email(email) do
+  def get_by_email(email) when is_binary(email) do
```

Comment:
**question:** what's the reasoning behind adding the `when is_binary(email)` guards here and in `upsert/2`? Just to make sure that we raise an exception if we somehow get a user without email information trying to log in rather than creating a new DB entry with a null email?

Reply:
Yes, exactly! An exception would otherwise be thrown from `String.downcase(email)` if email is nil, but I think it would be clearer to raise that kind of error from the top of the function.